probe_src/libprobe/generated/libc_hooks.c

```diff
@@ -1,5 +1,93 @@
+#ifdef __APPLE__
+#define F_SETSIG 0
+#define F_SETLEASE 0
+#define F_NOTIFY 0
+#define F_SETPIPE_SZ 0
+#define F_ADD_SEALS 0
+#define F_GETOWN_EX 0
+#define F_SETOWN_EX 0
+#define F_GET_RW_HINT 0
+#define F_SET_RW_HINT 0
+#define F_GET_FILE_RW_HINT 0
+#define F_SET_FILE_RW_HINT 0
+#define thrd_success 0
+#define thrd_nomem 1
+
+#define ftruncate64 ftruncate
+#define truncate64 truncate
+#define unwrapped_open64 unwrapped_open
+#define unwrapped_openat64 unwrapped_openat
+static pid_t (*unwrapped__Fork)(void);
+typedef off_t off64_t;
+typedef struct stat stat64_t;
+#endif
+
+typedef int (*thrd_start_t)(void *);
+
+#ifdef __APPLE__
+typedef pthread_t thrd_t;
+typedef struct {
+    thrd_start_t func;
+    void *arg;
+} thrd_wrapper_args_t;
+
+void *thrd_start_wrapper(void *arg) {
+    thrd_wrapper_args_t *wrapper_args = (thrd_wrapper_args_t *)arg;
+    int res = wrapper_args->func(wrapper_args->arg);
+    free(wrapper_args);
+    return (void *)(intptr_t)res;
+}
+
+static int unwrapped_thrd_create(thrd_t *thr, thrd_start_t func, void *arg) {
+    thrd_wrapper_args_t *wrapper_args = malloc(sizeof(thrd_wrapper_args_t));
+    if (wrapper_args == NULL) {
+        return thrd_nomem;
+    }
+    wrapper_args->func = func;
+    wrapper_args->arg = arg;
+    int ret = pthread_create(thr, NULL, thrd_start_wrapper, wrapper_args);
+    if (ret != 0) {
+        free(wrapper_args);
+    }
+    return (ret == 0) ? thrd_success : ret;
+}
+
+static pid_t unwrapped_clone(int (*fn)(void *), void *child_stack, int flags, void *arg, ...) {
+    errno = ENOSYS;
+    return -1;
+}
+#endif
+
+#ifdef __linux__
+int truncate64(const char *name, off64_t length) {
+    return truncate(name, length);
+}
+#endif
+
+#ifdef __APPLE__
+static ssize_t unwrapped_getdents64(int fd, void *dirp, size_t count) {
+    errno = ENOSYS;
+    return -1;
+}
+static int unwrapped_close_range(unsigned int lowfd, unsigned int maxfd, int flags) {
+    errno = ENOSYS;
+    return -1;
+}
+#endif
+
+#ifdef __linux__
+ssize_t getdents64(int fd, void *buffer, size_t length) {
+}
+#endif
+
```

Comment:
This file is generated. Desired changes should be managed in two ways: Linux and Mac specific `#define`s can be put in their own file and included in `lib.c` prior to the inclusion of this file.

Function-pointers that don't need to be interposed on Mac (e.g., `dup3`) should be switched off in `generator/gen_libc_hooks.py`.

Reply:
incorporated changes ( removed from `libc_hooks.c` and moved to OS specific headers, which are included in `lib.c` before `libc_hooks.c` is included).